edx_event_bus_kafka/internal/tests/test_consumer.py

```diff
@@ -305,6 +305,41 @@ def raise_exception():
         assert mock_emit.call_args_list == [call(self.normal_message)] * exception_count
         assert exc_info.value.args == ("Too many consecutive errors, exiting (4 in a row)",)
 
+    @override_settings(
+        EVENT_BUS_KAFKA_SCHEMA_REGISTRY_URL='http://localhost:12345',
+        EVENT_BUS_KAFKA_BOOTSTRAP_SERVERS='localhost:54321',
+        EVENT_BUS_TOPIC_PREFIX='dev',
+    )
+    def test_connection_reset_if_unusable(self):
+        """Confirm we reconnect to the database if our connection is deemed unusable"""
+        mock_emit = Mock(side_effect=side_effects([self.event_consumer._shut_down]))  # pylint: disable=protected-access
+        with patch.multiple(self.event_consumer,
```

Comment:
I get so lost in this. :) Is it possible to just patch what we need in a decorator on the method, and leave this complicated code for breaking out of the loop alone, so it doesn't get even more complicated?

Reply:
[fixed]